File: doc/api/ruby/example_command_line.rb
#!/usr/bin/env ruby
# vim: set sw=4 sts=4 et tw=80 :

=begin description
Basic command line handling for most examples.
=end

require 'getoptlong'
require 'singleton'
require 'Paludis'

class ExampleCommandLine < GetoptLong
    include Singleton

    def initialize
        super(
            [ '--log-level',             GetoptLong::REQUIRED_ARGUMENT ],
            [ '--environment',     '-E', GetoptLong::REQUIRED_ARGUMENT ]
        )

        @environment = ""
        each do | opt, arg |
            case opt
            when '--log-level'
                Paludis::Log.instance.log_level = Paludis::LogLevel::Debug
            when '--environment'
                @environment = arg
            end
        end
    end

    attr_reader :environment
end

File: doc/api/python/example_command_line.py
#!/usr/bin/env python
# vim: set fileencoding=utf-8 sw=4 sts=4 et :

"""Basic command line handling for most examples"""

import optparse
import paludis

class ExampleCommandLine:
    def __init__(self):
        parser = optparse.OptionParser()
        parser.add_option("", "--log-level", type = "choice",
                choices = ["debug", "qa", "warning", "silence"],
                action = "store", dest = "log_level", help = "Specify the log level")
        parser.add_option("-E", "--environment", action = "store", dest = "environment", default = "",
                help = "Environment specification")

        (options, args) = parser.parse_args()

        print options.log_level
        if options.log_level:
            paludis.Log.instance.log_level = getattr(paludis.LogLevel, options.log_level.upper())

if __name__ == "__main__":
    ExampleCommandLine()
